# Практическая работа №3: Обработка выборочных данных. Нахождение интервальных оценок параметров распределения. Проверка статистической гипотезы о нормальном распределении

Выполнили студенты гр. 0383 Желнин Максим, Рудакова Юлия, Сергевнин Дмитрий. Вариант №12


## Цель работы
Получение практических навыков вычисления интервальных статистических оценок параметров распределения выборочных данных и проверки «справедливости» статистических гипотез

## Основные теоретические положения

Интервальная статистическая оценка - оценка задаваемая двумя концами интервала в котором может находиться значение параметра $θ$

Точность статистической оценки параметра $θ$ - значение δ, которое удовлетворяет неравенству: $$δ > |θ^* - θ|$$

Вероятность $γ$ с которой выполняется неравенство $γ = P\{δ > |θ^* - θ|\} $ называют надежностью оценки $θ^*$

Доверительный интервал $$(θ^* - δ ; θ^* + δ)$$

Значение $α = 1 - γ$ - уровень значимости.

Статистической гипотезой называют гипотезу о виде неизвестного распределения исследуемой случайной величины или о значении параметров известных распределений.

Выдвинутую гипотезу  $H_0$ называют нулевой или основной гипотезой.

Противоречащая основной - конкурирующая $H_1$.

Критерий $χ^2$. Наблюденное значение случайной величины вычисляется на основании выборочных данных
$$χ^2_{набл} = \sum^K_{i=1} \frac{(n_i - n_i')^2}{n_i'} (5)$$

## Постановка задачи

Для заданной надежности определить (на основании выборочных данных и результатов выполнения практической работы №2) границы доверительных интервалов для математического ожидания и среднеквадратичного отклонения случайной величины. Проверить гипотезу о нормальном распределении исследуемой случайной величины с помощью критерия Пирсона $χ^{2}$
. Дать содержательную интерпретацию полученным результатам.

## Выполнение работы


### 1.
Вычислить точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении при заданном объёме выборки для доверительной точности γ∈{0.95,0.99}. Сделать выводы.

Перенесём данные из прошлых работ.

In [1]:
import numpy as np
from scipy.special import erf

In [2]:
laplas = lambda x: erf(x/2**0.5)/2

In [37]:
N = 111
X_avg = 128.536
S = 24.3566
y_1 = 0.95
y_2 = 0.99
t_accuracy_1 = 1.9855;
t_accuracy_2 = 2.627;

t = 1.9855 при γ = 0.95 и n = N-1 = 110

In [38]:
accuracy_1 = t_accuracy_1 * S / np.sqrt(N)
a_left_1 = X_avg - accuracy_1
a_right_1 = X_avg + accuracy_1
print("Доверительный интервал для математического ожидания при y=0.95 (", a_left_1, " ; ", a_right_1, ")")
print("Точность ", accuracy_1)
accuracy_2 = t_accuracy_2 * S / np.sqrt(N)
a_left_2 = X_avg - accuracy_2
a_right_2 = X_avg + accuracy_2
print("Доверительный интервал для математического ожидания при y=0.99 (", a_left_2, " ; ", a_right_2, ")")
print("Точность ", accuracy_2)

Доверительный интервал для математического ожидания при y=0.95 ( 123.94586915150799  ;  133.126130848492 )
Точность  4.590130848492013
Доверительный интервал для математического ожидания при y=0.99 ( 122.46283266734399  ;  134.60916733265603 )
Точность  6.0731673326560145


Вывод:

Мат. ожидание с вероятностью 0.95 в интервале ( 123.94586915150799  ;  133.126130848492 )

Мат. ожидание с вероятностью 0.99 в интервале ( 122.46283266734399  ;  134.60916733265603 )

Второй интервал более вероятен, но и имеет больший диапозон, следовательно не всегда информативен

2. Для вычисления границ доверительного интервала для среднеквадратичного отклонения определить значение q при заданных γ и  n. Построить доверительные интервалы, сделать выводы.

In [8]:
q = 0.143
sd_left = S*(1-q)
sd_right = S*(1+q)
print("q = ", q)
print("Доверительный интервал для среднеквадратического отклонения: ")
print(sd_left, sd_right)

q =  0.143
Доверительный интервал для среднеквадратического отклонения: 
20.8736062 27.8395938


Из таблицы q = 0.143


Вывод:

С вероятностью 0.95 СКВО лежит в интервале (20.8736062; 27.8395938)

3.
Проверить гипотезу о нормальности заданного распределения с помощью критерия χ2 (Пирсона). Для этого необходимо найти теоретические частоты и вычислить наблюдаемое значение критерия. Для удобства вычисления необходимо заполнить табл. 1.

H0 - гипотеза о том, что выборка имеет нормальное распределение, H1 - альтернативная гипотеза, что выборка имеет не нормальное распределение.

In [45]:
freq = [13, 13, 13, 19, 18, 17, 7, 6]
mid_values = [84, 102, 114, 126, 138, 150, 162, 174]
borders = [[72.6, 96.6, 108.7, 120.7, 132.8, 144.8, 156.9, 168.9],[96.6, 108.7, 120.7, 132.8, 144.8, 156.9, 168.9, 181.0]]
z = [[(e - X_avg)/S for e in borders[0]],[(i - X_avg)/S for i in borders[1]]]
z[0][0] = -float('inf')
z[1][-1] = float('inf')
p = []
for i in range(0, len(z[0])):
    p.append(laplas(z[1][i])-laplas(z[0][i]))
freq_theor = [x*sum(freq) for x in p]

In [46]:
print("(x_i-1, x_i):")
for i in range(len(borders[0])):
  print(borders[0][i], borders[1][i])
print("n_i")
print(freq)
print("p_i")
print(p)
print("n_i'")
print(freq_theor)
print("(n_i-n_i')^2")
print([(freq_theor[i] - freq[i])**2 for i in range(len(freq))])
print("(n_i-n_i')^2/n_i'")
print([(freq_theor[i] - freq[i])**2 / freq_theor[i] for i in range(len(freq))])
print("n_i^2")
print([freq[i]**2 for i in range(len(freq))])
print("n_i^2/n_i'")
print([freq[i]**2 / freq_theor[i] for i in range(len(freq))])

(x_i-1, x_i):
72.6 96.6
96.6 108.7
108.7 120.7
120.7 132.8
132.8 144.8
144.8 156.9
156.9 168.9
168.9 181.0
n_i
[13, 13, 13, 19, 18, 17, 7, 6]
p_i
[0.09489769385540875, 0.11281040228097877, 0.16612439957590025, 0.19565341522520005, 0.17836592704796528, 0.13004358703510846, 0.07336605773131383, 0.04873851724812461]
n_i'
[10.059155548673328, 11.95790264178375, 17.609186355045427, 20.739262013871205, 18.90678826708432, 13.784620225721497, 7.776802119519266, 5.166282828301209]
(n_i-n_i')^2
[8.648566086898876, 1.0859669040012865, 21.24459885553695, 3.0250323528953187, 0.8222649613217835, 10.338667092839277, 0.6034215328896234, 0.6950843223854317]
(n_i-n_i')^2/n_i'
[0.8597705885997071, 0.09081583422553219, 1.206449771567662, 0.14586017336933504, 0.043490462245949074, 0.7500146484665408, 0.07759250185562454, 0.1345424448266204]
n_i^2
[169, 169, 169, 361, 324, 289, 49, 36]
n_i^2/n_i'
[16.80061503992638, 14.132913192441782, 9.597263416522235, 17.40659815949813, 17.13670219516163, 20.965394422745

k = K - 3 = 5

$χ2_{крит} = 11.1$ из таблицы

In [47]:
k = 5
chi2_expected = 11.1

In [48]:
stats_crit = []
for i in range(len(freq)):
    stats_crit.append((freq[i] - freq_theor[i])**2/freq_theor[i])
chi2_calculated = sum(stats_crit)
print("Наблюдаемое значение критерия: ", chi2_calculated)
print("Наблюдаемое < критическое: ", chi2_calculated < chi2_expected)

Наблюдаемое значение критерия:  3.3085364251569715
Наблюдаемое < критическое:  True


Основная гипотеза подтверждена

4.

$χ^2_{набл} = \sum\frac{(n_i - n^`_i)^2}{n_i^`} = \sum\frac{n_i^2 - 2n_in^`_i + (n^`_i)^2}{n_i^`} = \sum\frac{n_i^2}{n_i^`} - 2\sum n_i + \sum n^`_i = \sum\frac{n_i^2}{n_i^`} - 2N + N = \sum\frac{n_i^2}{n_i^`} - N$

In [50]:
print(sum([(freq[i]**2) / freq_theor[i] for i in range(len(freq))]) - sum(freq))

3.308536425156987


Сошлось с посчитанным ранее

5. Сделано в п.3, так как наблюдаемое значение критерия ниже критического, то гипотеза о нормальности распределения подтверждена.

##Вывод


В процессе выполнения практической работы были рассчитаны точность и доверительный интервал для математического ожидания при неизвестном стандартном отклонении для уровней доверия 0.95, 0.99, а также было определено значение q. Потом при помощи критерия $χ^2$ была проверена гипотеза о типа распределения выборки и было получено, что выборка имеет нормальное распределение. Для этой проверки были вычислены теоретические частоты и определено наблюдаемое значение критерия. Было также доказано равенство двух формул для расчета $χ^2_{набл}$.